library/include/DataIdentity.h

```diff
@@ -712,6 +780,30 @@ namespace df
     }
 #endif
 
+    template<isIndexedContainer C>
+    struct DFHACK_EXPORT identity_traits<C> {
+        static type_identity* get() {
+            static generic_container_identity<C> identity{};
+            return &identity;
+        }
+    };
+
+    template<isAssocContainer C>
+    struct DFHACK_EXPORT identity_traits<C> {
+        static type_identity* get() {
+            static generic_assoc_container_identity<C> identity{};
+            return &identity;
+        }
+    };
+
+    template<isSetContainer C>
+    struct DFHACK_EXPORT identity_traits<C> {
+        static type_identity* get() {
+            static generic_set_container_identity<C> identity{};
+            return &identity;
+        }
+    };
+
```

Comment:
some context here: https://discord.com/channels/793331351645323264/807443262704058408/1239888219764690986

Reply:
i'm still exploring this - i think it has to do inconsistent typing of the `get` static method. there appears to be no reason why all of the `get` methods cannot return a `type_identity*` as the type is polymorphic anyhow

however, i'm still investigating this

Reply:
the issue of duplicate implementations still exists, but in this case the static object being created is read-only so if there are two copies there should be no risk of harm. the problem that arises when there are multiple instances of the same specialization is when there is mutable data, but there's no mutable data in a `type_identity`

also, both gcc and msvc have improved their loading models since 2012 to reduce problems with duplicated implementations

there is a _ton_ of stuff in our code that is there to work around compiler bugs and shortcomings, or even C++ language shortcomings, that were fixed, in some cases, over a decade ago